client/data/deposits/resolvers.js

```diff
@@ -30,6 +30,11 @@ import { formatDateValue } from 'utils';
  * @param {string} id Identifier for specified deposit to retrieve.
  */
 export function* getDeposit( id ) {
+	// Validate input to avoid path traversal request.
+	if ( /[.\/?#&]/.test( id ) ) {
+		return;
+	}
```

Comment:
Did you consider being stricter here? As we're dealing with stripe ids in a strict format. Restricting by non-word characters could make this more readable and deny anything unexpected.
```suggestion
    // Avoid lookup if the id contains any unexpected characters.
	if ( /\W/.test( id ) ) {
		return;
	}
```

Reply:
Good idea! Thank you for suggesting that.

Reply:
+1 to this, thanks for making this change.